packages/shared/src/hooks/chat/useChatStream.ts

```diff
@@ -0,0 +1,168 @@
+import { useRef, useState, useCallback, useEffect, useContext } from 'react';
+import { useQueryClient } from 'react-query';
+import { useAuthContext } from '../../contexts/AuthContext';
+import {
+  SearchChunk,
+  updateSearchData,
+  initializeSearchSession,
+  SearchChunkError,
+  sendSearchQuery,
+  Search,
+} from '../../graphql/search';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+import {
+  UseChatMessage,
+  UseChatMessageType,
+  CreatePayload,
+  SourcesMessage,
+  TokenPayload,
+  UseChatStream,
+} from './types';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+import { AnalyticsEvent } from '../../lib/analytics';
+
+export const useChatStream = (): UseChatStream => {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const { user, accessToken } = useAuthContext();
+  const client = useQueryClient();
+  const sourceRef = useRef<EventSource>();
+  const [sessionId, setSessionId] = useState<string>();
+
+  const executePrompt = useCallback(
+    async (value: string) => {
+      if (!value) {
+        return;
+      }
+
+      if (sourceRef.current?.OPEN) {
+        sourceRef.current.close();
+      }
+
+      setSessionId(undefined);
```

Comment:
undefined or null?

Reply:
yeah, `null` is probably better since we are using it in queryKey.